rng_tools/src/generators/gen3/wild/generator.rs

```diff
@@ -0,0 +1,208 @@
+use crate::gen3::EncounterSlot;
+use crate::gen3::Gen3Ability;
+use crate::gen3::Gen3Lead;
+use crate::gen3::Gen3Method;
+use crate::gen3::ShinyType;
+use crate::rng::lcrng::Pokerng;
+use crate::rng::{Rng, StateIterator};
+use crate::{Gender, GenderRatio, Nature, PkmFilter, gen3_shiny};
+use crate::{IvFilter, Ivs};
+use itertools::Itertools;
+use wasm_bindgen::prelude::wasm_bindgen;
+
+pub struct Gen3WOpts {
+    shiny_type: Option<ShinyType>,
+    ability: Option<Gen3Ability>,
+    gender: Option<Gender>,
+    nature: Option<Nature>,
+    iv_range: (Ivs, Ivs),
+    tid: u16,
+    sid: u16,
+    gender_ratio: GenderRatio,
+    encounter_slot: Option<EncounterSlot>,
+    method: Option<Gen3Method>,
+    min_advances: usize,
+    max_advances: usize,
+}
+
+#[derive(Debug, PartialEq)]
+pub struct GeneratedPokemon {
+    pid: u32,
+    shiny: bool,
+    ability: Gen3Ability,
+    gender: Gender,
+    ivs: Ivs,
+    nature: Nature,
+    advances: usize,
+}
+
+/// generate pokemon -> call encounter first, then call rng for pid creation and then rng calls for ivs. all filters after that just reference the pid to see if pokemon matches
+/// what user has submitted as wanting
+
+pub fn generate_pokemon(rng: &mut Pokerng, settings: &Gen3WOpts) -> Option<GeneratedPokemon> {
+    let encounter_rand = (rng.rand::<u32>() >> 8) as u8;
+    let encounter_slot = EncounterSlot::from_rand(encounter_rand);
+
+    if !EncounterSlot::passes_filter(settings.encounter_slot, encounter_slot) {
+        println!("encounter - Current seed: {}", rng.seed());
+        return None;
+    }
+
```

Comment:
@sorendog It looks like there's a missing RNG call to get the Pokemon's level in between the encounter slot and the PID.

When I add an extra rand call to this line:
```suggestion
rng.rand::<u32>(); // level
```

I get this result from your test:
```
GeneratedPokemon {
  pid: 0xfc3367db,
  shiny: false,
  ability: Ability1,
  gender: Male,
  ivs: Ivs {
    hp: 12,
    atk: 25,
    def: 27,
    spa: 2,
    spd: 31,
    spe: 30
  },
  nature: Bold,
  advances: 5
}
```

Which perfectly matches PokeFinder:
<img width="948" alt="Screenshot 2025-05-14 at 6 44 58 PM" src="https://github.com/user-attachments/assets/3aa83421-706b-4ab3-8950-1e786160552c" />

Reply:
Thank you!! The problem I'm having now is that the advances don't match, that says advance 5 but pokefinder shows it as 0. 

But maybe I just need to step away from it for a little bit and I'll figure that out on my own eventually.